addons/sale/models/sale_order.py

```diff
@@ -864,6 +867,21 @@ def action_quotation_send(self):
             'context': ctx,
         }
 
+    def action_send_email(self):
+        self.ensure_one()
+        report_action = self.action_quotation_send()
+        if self.env.is_admin() and not self.env.company.external_report_layout_id and not self.env.context.get('discard_logo_check'):
+            return self._action_configure_external_report_layout(report_action)
+        return report_action
+
+    def _action_configure_external_report_layout(self, report_action):
+        action = self.env["ir.actions.actions"]._for_xml_id("web.action_base_document_layout_configurator")
+        py_ctx = json.loads(action.get('context', {}))
+        py_ctx['report_action'] = report_action
+        py_ctx['dialog_size'] = 'extra-large'
+        action['context'] = py_ctx
```

Comment:
Use `.update` function dict to update multiple attribute at once
Also use should use `safe_eval` instead of `json` :thinking: 
(I am not sure which one is better but `safe_eval` is in odoo.tools)